### PR TITLE
Slim 1387 Successful exchange encryption key on new m bus device

### PR DESCRIPTION
**Pull request by bjornamkreutz**
SLIM-1387, test [@SmartMeteringConfiguration]

### CODE REVIEW

File: cucumber-tests-platform-smartmetering/src/test/resources/features/osgp-adapter-ws-smartmetering/SmartMeteringConfiguration.feature [hunk 1]
```diff
@@ -97,7 +97,23 @@ Feature: SmartMetering Configuration
...
+      | DeviceIdentification | TEST2560000000001 |
+      | DeviceType           | SMART_METER_E     |
+    And a dlms device without user key
+      | DeviceIdentification        | TESTG102411111111 |
+      | DeviceType                  | SMART_METER_G     |
+      | GatewayDeviceIdentification | TEST2560000000001 |
+      | Channel                     |                 1 |
+    When the exchange user key request is received
+      | DeviceIdentification | TESTG102411111111 |
+    Then the new user key is generated and should be set on the gas device
```
Comment (smvdheijden) [inline cucumber-tests-platform-smartmetering/src/test/resources/features/osgp-adapter-ws-smartmetering/SmartMeteringConfiguration.feature]: M-Bus device

File: cucumber-tests-platform-smartmetering/src/test/java/com/alliander/osgp/cucumber/platform/smartmetering/glue/steps/ws/smartmetering/smartmeteringconfiguration/SetEncryptionKeyExchangeOnGMeter.java [hunk 2]
```diff
@@ -67,5 +79,40 @@ public void theNewUserKeyShouldBeSetOnTheGasDevice(final Map<String, String> set
...
+                .findByDlmsDeviceAndSecurityKeyTypeAndValidToNotNull(dlmsDevice, SecurityKeyType.G_METER_ENCRYPTION);
+
+        final SecurityKey newSecurityKey = this.dlmsSecurityKeyRepository
+                .findByDlmsDeviceAndSecurityKeyTypeAndValidToIsNull(dlmsDevice, SecurityKeyType.G_METER_ENCRYPTION);
+
+        assertNotNull("End date of generated user key is null", oldSecurityKey.getValidTo());
+        assertNotNull("Generated user key is null", newSecurityKey);
     }
+
+    @Then("^the new user key is generated and should be set on the gas device$")
```
Comment (smvdheijden) [inline cucumber-tests-platform-smartmetering/src/test/java/com/alliander/osgp/cucumber/platform/smartmetering/glue/steps/ws/smartmetering/smartmeteringconfiguration/SetEncryptionKeyExchangeOnGMeter.java]: M-Bus device

File: cucumber-tests-platform-smartmetering/src/test/java/com/alliander/osgp/cucumber/platform/smartmetering/glue/steps/ws/smartmetering/smartmeteringconfiguration/SetEncryptionKeyExchangeOnGMeter.java [hunk 3]
```diff
@@ -67,5 +79,40 @@ public void theNewUserKeyShouldBeSetOnTheGasDevice(final Map<String, String> set
...
+    public void theNewUserKeyIsGeneratedAndShouldBeSetOnTheGasDevice(final Map<String, String> settings)
```
Comment (smvdheijden) [inline cucumber-tests-platform-smartmetering/src/test/java/com/alliander/osgp/cucumber/platform/smartmetering/glue/steps/ws/smartmetering/smartmeteringconfiguration/SetEncryptionKeyExchangeOnGMeter.java]: GasDevice -> MBusDevice

**Comment by bjornamkreutz** [comment]
retest this please

**Pull request merged**